smac/tae/execute_func.py

```diff
@@ -109,3 +95,79 @@ def run(self, config, instance=None,
         runtime = float(obj.wall_clock_time)
 
         return status, cost, runtime, additional_run_info
+
+    def _call_ta(self, obj, config, instance, seed):
+        raise NotImplementedError()
+
+
+class ExecuteTAFunc(AbstractTAFunc):
+
+    """Evaluate function for given configuration and resource limit.
+
+    Parameters
+    ----------
+    ta : callable
+        Function (target algorithm) to be optimized. Needs to accept at least a
+        configuration and a seed. Can return a float (the loss) or a tuple
+        (the loss and additional run information in a dictionary).
+    stats : smac.stats.stats.Stats
+        Stats object to collect statistics about runtime etc.
+    run_obj: str
+        Run objective (runtime or quality)
+    runhistory: RunHistory
+        runhistory to keep track of all runs; only used if set
+    par_factor: int
+        Penalized average runtime factor. Only used when `run_obj='runtime'`
+    """
+    
+    def __init__(self, ta, stats=None, runhistory=None, run_obj="quality", par_factor=1):
+        super().__init__(ta=ta, stats=stats, runhistory=runhistory, run_obj=run_obj, par_factor=par_factor)
+        self._supports_memory_limit = True
+
+    def _call_ta(self, obj, config, instance, seed):
+
+        if instance:
+            rval = obj(config, instance, seed)
+        else:
+            rval = obj(config, seed)
+
+        if isinstance(rval, tuple):
+            result = rval[0]
+            additional_run_info = rval[1]
+        else:
+            result = rval
+            additional_run_info = {}
+
+        return result, additional_run_info
+
+
+class ExecuteTAFunc4FMIN(AbstractTAFunc):
```

Comment:
If I knew a better name I would immediately change this. What about ExecuteTaFuncSimple?